pytests/func/test_run.py

```diff
@@ -12,24 +12,37 @@
 from pytests.helpers import print
 from pytests.helpers.pattern import ANY
 from pytests.helpers.timeline import Event
-from pytests.helpers.session import START_METHOD_LAUNCH, START_METHOD_CMDLINE
+from pytests.helpers.session import START_METHOD_LAUNCH, START_METHOD_CMDLINE, START_METHOD_IMPORT
 
 
 @pytest.mark.parametrize('run_as', ['file', 'module', 'code'])
-@pytest.mark.parametrize('start_method', [START_METHOD_LAUNCH, START_METHOD_CMDLINE])
+@pytest.mark.parametrize('start_method', [START_METHOD_LAUNCH, START_METHOD_CMDLINE, START_METHOD_IMPORT])
 def test_run(debug_session, pyfile, run_as, start_method):
     @pyfile
     def code_to_debug():
         import os
         import sys
         import backchannel
 
+        if os.getenv('PTVSD_ENABLE_ATTACH', False):
```

Comment:
This should probably go into `DebugSession.initialize()`, so we can run other tests with import too without repeating it every time. The session can generate another temp script with `@pyfile` that does the import, and then invokes `target[1]` via the `runpy` module if `target[0]` is "file" or "module". For "code", we can just prepend "import ptvsd; ...;" directly to `target[1]`.

Reply:
That is a good idea, will update.